Surface_mesh/include/CGAL/Surface_mesh/IO.h

```diff
@@ -267,51 +267,12 @@ bool read_off_ascii(Surface_mesh<Point_3>& mesh,
 template <typename K>
 bool read_off(Surface_mesh<K>& mesh, const std::string& filename)
 {
-  std::string  line;
-    bool  has_texcoords = false;
-    bool  has_normals   = false;
-    bool  has_hcoords   = false;
-    bool  has_dim       = false;
-    bool  is_binary     = false;
-
-    // open file (in ASCII mode)
-    std::ifstream in(filename.c_str());
-    if (!in) return false;
-
-    // read header: [ST][C][N][4][n]OFF BINARY
-    std::getline(in,line);
-    const char *c = line.c_str();
-    if (c[0] == 'S' && c[1] == 'T') { has_texcoords = true; c += 2; }
-    if (c[0] == 'N') { has_normals = true; ++c; }
-    if (c[0] == '4') { has_hcoords = true; ++c; }
-    if (c[0] == 'n') { has_dim     = true; ++c; }
-    if (strncmp(c, "OFF", 3) != 0) { in.close(); return false; } // no OFF
-    if (strncmp(c+4, "BINARY", 6) == 0) is_binary = true;
-
-
-    // homogeneous coords, and vertex dimension != 3 are not supported
-    if (has_hcoords || has_dim)
-    {
-        in.close();
-        return false;
-    }
-
-
-    // if binary: reopen file in binary mode
-    if (is_binary)
-    {
-      in.close();
-      in.open(filename.c_str(), std::ios::binary);
-      std::getline(in,line);
-    }
-
-    // read as ASCII or binary
-    bool ok = (is_binary ?
-               internal::read_off_binary(mesh, in, has_normals, has_texcoords) :
-               internal::read_off_ascii(mesh, in, has_normals, has_texcoords));
+  std::ifstream in(filename.c_str());
+  if (in.fail())
+    return false;
+  in >> mesh;
```

Comment:
The new code certainly cannot read binary file (because of the `ifstream` open mode. That seems to be a regression.

Reply:
Indeed. What should be done here? The real bug was only in the write function but I though that maybe we could get rid of duplicated code. I can simply undo the change for the read function.

Reply:
I think you should indeed undo the change for the read function, otherwise the demo will no longer read the colors of colored OFF.

Reply:
Amended